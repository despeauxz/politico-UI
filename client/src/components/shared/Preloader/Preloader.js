/* eslint-disable max-len */
import React from 'react';
import Loader from 'react-loader-spinner';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './Preloader.scss';

const Preloader = ({ type, style, height, width, color, ...rest }) => {
    const className = classNames({
        'main-preloader center justify-center items-center': type === 'page',
        '': type === 'button'
    });

    return (
        <div className={className} {...rest}>
            <Loader type={style} height={height} width={width} color={color} className="text-center mx-auto center" />
        </div>
    );
};

Preloader.propTypes = {
    type: PropTypes.string,
    style: PropTypes.string,
    height: PropTypes.number,
    width: PropTypes.number,
    color: PropTypes.string,
};

Preloader.defaultProps = {
    type: null,
    style: null,
    height: null,
    width: null,
    color: null
};

export default Preloader;
