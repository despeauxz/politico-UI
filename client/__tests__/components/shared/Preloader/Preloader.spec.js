import React from 'react';
import Preloader from '@components/shared/Preloader/Preloader';


const props = {
    type: 'button',
    height: 40,
    width: 40,
    style: 'TailSpin',
    color: 'green'
};

describe('<Preloader />', () => {
    it('should render a preloader with type ~ button', () => {
        const shallowWrapper = mount(<Preloader {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Preloader')).toBeTruthy();
    });

    it('should render a preloader with type ~ main-page', () => {
        const shallowWrapper = mount(<Preloader { ...props } type="main-preloder" />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Preloader')).toBeTruthy();
        expect(shallowWrapper.find('main-preloder')).toBeTruthy();
    });
});
