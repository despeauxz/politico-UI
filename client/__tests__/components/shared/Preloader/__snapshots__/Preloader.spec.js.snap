// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> should render a preloader with type ~ button 1`] = `
<div
  className=""
>
  <Loader
    color="green"
    height={40}
    type="TailSpin"
    width={40}
  />
</div>
`;

exports[`<Preloader /> should render a preloader with type ~ main-page 1`] = `
<div
  className=""
>
  <Loader
    color="green"
    height={40}
    type="TailSpin"
    width={40}
  />
</div>
`;
